src/api/v1alpha1/edgeconnect/edgeconnect_types.go

```diff
@@ -0,0 +1,157 @@
+	// +required
+	// +kubebuilder:validation:Required
+	ApiServer string `json:"apiServer"`
+
+	// EdgeConnect uses the OAuth client to authenticate itself with the Dynatrace platform.
+	// +required
+	// +kubebuilder:validation:Required
+	OAuth OAuthSpec `json:"oauth"`
+
+	// Optional: restrict outgoing HTTP requests to your internal resources to specified hosts
+	// +kubebuilder:validation:Optional
+	HostRestrictions string `json:"hostRestrictions,omitempty"`
+
+	// Optional: image reference
+	// +kubebuilder:validation:Optional
+	ImageRef ImageRefSpec `json:"imageRef,omitempty"`
+
+	// Optional: enables automatic restarts of EdgeConnect pods in case a new version is available (the default value is: true)
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	AutoUpdate bool `json:"autoUpdate"`
+
+	// Optional: pull secret for your private registry
+	// +kubebuilder:validation:Optional
+	CustomPullSecret string `json:"customPullSecret,omitempty"`
+
+	// Optional: adds additional annotations for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Optional: adds additional labels for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Optional: adds additional environment variables for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Amount of replicas for your EdgeConnect (the default value is: 1)
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=1
+	Replicas *int32 `json:"replicas"`
+
+	// Optional: define resources requests and limits for single pods
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Optional: node selector to control the selection of nodes for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Optional: set tolerations for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Optional: set topology spread constraints for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+}
+
+type OAuthSpec struct {
+	// The secret from your OAuth client generation
+	// +required
+	// +kubebuilder:validation:Required
+	ClientSecret string `json:"clientSecret"`
+	// The token endpoint URL
+	// +required
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+	// URN identifying your account. You get the URN when creating the OAuth client
+	// +required
+	// +kubebuilder:validation:Required
+	Resource string `json:"resource"`
+}
+
+type ImageRefSpec struct {
+	// Optional: if specified, indicates the EdgeConnect repository to use
```

Comment:
```suggestion
	// Custom EdgeConnect image repository
```

"Optional" and "if specified" is redundant.

Reply:
We could use the `+kubebuilder:example` marker to show what an expected value for this field (and some others where it makes sense) is, e.g. `docker.io/edgeconnect`